### PR TITLE
Cloud Run and Cloud Functions integration with Firebase Hosting hashicorp/terraform-provider-google#12955

### PR DESCRIPTION
**Pull request by rainshen49**
https://github.com/hashicorp/terraform-provider-google/issues/12955

Support the integration with Cloud Run and Cloud Functions. Learn more in [Firebase Hosting Serverless Documentation](https://firebase.google.com/docs/hosting/serverless-overview). Static site hosting is not supported in Terraform at the moment. 

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_firebase_hosting_version
google_firebase_hosting_release
```

### CODE REVIEW

**Comment by c2thorn** [comment]
apologies @rainshen49, but I think this needs to be rebased _again_. We had an unexpected build-breaking PR that we have recently reverted.

**Comment by rainshen49** [comment]
OK, rebased

File: mmv1/products/firebasehosting/terraform.yaml [hunk 1]
```diff
@@ -63,6 +63,89 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
         test_env_vars:
           project_id: :PROJECT_NAME
+  Release: !ruby/object:Overrides::Terraform::ResourceOverride
+    import_format: [
+      'sites/{{site_id}}/releases/{{release_id}}',
+      'sites/{{site_id}}/channels/{{channel_id}}/releases/{{release_id}}',
+      '{{site_id}}/{{release_id}}',
+      '{{site_id}}/{{channel_id}}/{{release_id}}'
+    ]
+    skip_delete: true # can't delete a release
```
Comment (c2thorn) [inline mmv1/products/firebasehosting/terraform.yaml:73]: I want to confirm that we won't have orphaned resources from running these below examples as tests. They should be eventually deleted once the parent "site" is removed right?

Comment (rainshen49) [reply mmv1/products/firebasehosting/terraform.yaml:73]: Yes, and even for the default hosting site, which can't be deleted, there is a retainedReleaseCount field (default 10) that specifies automatically deletes the oldest release if full. https://firebase.google.com/docs/reference/hosting/rest/v1beta1/sites.channels#Channel.FIELDS.retained_release_count

**Review by c2thorn — APPROVED** [review]

**Pull request merged**